include/tuvx/linalg/linalg.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+namespace tuvx {
+namespace linalg {
+
+// tridiag matrix
+template <typename T> struct trid_mat {
+  int size;
+  std::vector<T> udiag; // upper diagonal
+  std::vector<T> ldiag; // lower diagonal
+  std::vector<T> mdiag; // main diagonal
+};
+
+template <typename T> void fill_rand_vec(std::vector<T> &x, int size);
+
+template <typename T> void fill_rand_mat(trid_mat<T> &trid_mat, int n);
+
+template <typename T> void print_vec(std::vector<T>);
+
+template <typename T> void print_trid_mat(trid_mat<T> x);
+
+template <typename T>
+std::vector<T> tridiag_solve(trid_mat<T> A, std::vector<T> b);
```

Comment:
Matt setup several other types `Array2D`, `Array3D` that we will be using to hold the bulk of the data for operational use in TUV (unless you make a stronger case for eigen or some other matrix library). Is this code expected to work with that type or would we do a slight refactor when we go to use this code?

Reply:
Since the tridiagonal matrix is sparse, I only store the diagonal values in three std::vectors  instead of the full matrix in `Array2D`, as the rest of the entries are zero. The function is tailored to work with this specific sparse representation as the algorithm is only designed for tridiagonal forms. Although I could refactor it to accept three `Array1D` arguments instead (`linpak%tridiag` is designed like this too).

Reply:
Ok, `std::vector` seems fine then. We can always refactor this later if we decide we need something fancier